src/providers/details/tabs/Credentials/components/CredentialsSectionEditMode.tsx

```diff
@@ -0,0 +1,121 @@
+import { type FC, type ReactNode, useState } from 'react';
+import { AlertMessageForModals } from 'src/modules/Providers/modals/components/AlertMessageForModals';
+import type { ValidationMsg } from 'src/modules/Providers/utils/validators/common';
+import { ValidationState } from 'src/providers/utils/types';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import type { IoK8sApiCoreV1Secret, V1beta1Provider } from '@kubev2v/types';
+import {
+  Button,
+  Divider,
+  Flex,
+  FlexItem,
+  HelperText,
+  HelperTextItem,
+} from '@patternfly/react-core';
+
+import { getCredentialsEditModeByType } from './utils/getCredentialsEditModeByType';
+import { getCredentialsValidatorByType } from './utils/getCredentialsValidatorByType';
+import { isSecretDataChanged } from './utils/isSecretDataChanged';
+import { patchSecretData } from './utils/patchSecretData';
+
+import './CredentialsSectionEditViewModes.style.css';
+
+type CredentialsSectionEditModeProps = {
+  provider: V1beta1Provider;
+  secret: IoK8sApiCoreV1Secret;
+  toggleEdit: () => void;
+};
+
+const CredentialsSectionEditMode: FC<CredentialsSectionEditModeProps> = ({
+  provider,
+  secret,
+  toggleEdit,
+}) => {
+  const { t } = useForkliftTranslation();
+  const [isLoading, setIsLoading] = useState(false);
+  const [dataChanged, setDataChanged] = useState(false);
+  const [changedSecret, setChangedSecret] = useState<IoK8sApiCoreV1Secret>(secret);
+  const [dataError, setDataError] = useState<ValidationMsg>({ type: ValidationState.Default });
+  const [alertMessage, setAlertMessage] = useState<ReactNode>();
+
+  const validator = getCredentialsValidatorByType(provider);
+  const CredentialsEditModeByType = getCredentialsEditModeByType(provider);
+
+  if (!secret?.data || !CredentialsEditModeByType || !validator) {
+    return <span className="text-muted">{t('No credentials found.')}</span>;
+  }
+  const setNewSecret = (newValue: IoK8sApiCoreV1Secret) => {
+    setChangedSecret(newValue);
+    setDataChanged(isSecretDataChanged(secret, newValue));
+    setDataError(validator(newValue, provider));
+    setAlertMessage(null);
+  };
+
+  const onCancel = () => {
+    setNewSecret(secret);
+    toggleEdit();
+  };
+
+  const onNewSecretChange = (newValue: IoK8sApiCoreV1Secret) => {
+    setNewSecret(newValue);
+  };
+
+  const onUpdate = async () => {
+    setIsLoading(true);
+
+    try {
+      await patchSecretData(changedSecret, true);
+      setDataChanged(false);
+
+      setIsLoading(false);
+      toggleEdit();
+    } catch (err: unknown) {
+      const errorMessage = err instanceof Error ? err.message : err?.toString();
+      setAlertMessage(<AlertMessageForModals title={t('Error')} message={errorMessage} />);
```

Comment:
we talked about it in a couple of pr's - save only the msg string, not the whole component.

Reply:
We set  title, message and variant as different props for the `AlertMessageForModals` so setting the component is easier, but since we only set an "error" title in current implementation then I'll set it as default.